### ISSUE TITLE
Effects are lost in legacy project

### ISSUE THREAD

**Issue by Flashwalker** [OpenShot/openshot-qt#5121]
**Describe the bug**
I have a project that was made in an old version of Openshot, GTK or QT, can't remember, and now the effects that were applied are gone.
There are errors in the log file about legacy project file.

**Steps to reproduce the behavior:**
1. Project file 'https://my.opendesktop.org/s/P3ewPXcDKCKSnjN'

**Expected behavior**
Is there a way to convert the project file to get effects back?

**System Details**
- OpenShot Version: 3.0.0
- libopenshot Version:  0.3.0
- Platform: Linux-6.0.6-76060006-generic-x86_64-with-glibc2.35
- Processor: x86_64
- Machine: x86_64
- Python version: 3.10.6
- Qt5 version: 5.15.3
- PyQt5 version: 5.15.6
- Qt Detected Languages: ['en']
- LANG Environment Variable: ru_RU.UTF-8
- LOCALE Environment Variable: 
- Daily Build: Skipped this step

**Log Files**
[openshot-qt.log](https://s3.amazonaws.com/logs.openshot.org/1l1vuqkwary964kgjwdogkfu8ex09sl0/openshot-qt.zip) (6 KB)

<details>

```
15:10:38 INFO app: ------------------------------------------------
15:10:38 INFO app:             Sat Jan 28 15:10:38 2023            
15:10:38 INFO app:               Starting new session              
15:10:38 INFO app: ------------------------------------------------
15:10:38 INFO app:             OpenShot (version 3.0.0)            
15:10:38 INFO app: ------------------------------------------------
15:10:38 INFO app: openshot-qt version: 3.0.0
15:10:38 INFO app: libopenshot version: 0.3.0
15:10:38 INFO app: platform: Linux-6.0.6-76060006-generic-x86_64-with-glibc2.35
15:10:38 INFO app: processor: x86_64
15:10:38 INFO app: machine: x86_64
15:10:38 INFO app: python version: 3.10.6
15:10:38 INFO app: qt5 version: 5.15.3
15:10:38 INFO app: pyqt5 version: 5.15.6
15:10:38 INFO project_data: Setting default profile to HD 720p 30 fps
15:10:38 INFO project_data: Apply default audio playback settings: 44100, 2 channels
15:10:38 INFO app: checking babl_ext_path: /usr/lib/python3/dist-packages/openshot_qt/lib/babl-ext
15:10:38 INFO language: Qt Detected Languages: ['ru', 'en']
15:10:38 INFO language: LANG Environment Variable: ru_RU.UTF-8
15:10:38 INFO language: LOCALE Environment Variable: 
15:10:38 INFO language: OpenShot Preference Language: Default
15:10:38 INFO app: Setting font to /usr/lib/python3/dist-packages/openshot_qt/images/fonts/Ubuntu-R.ttf
15:10:38 INFO app: Setting custom dark theme
15:10:38 INFO logger_libopenshot: Connecting to libopenshot with debug port: 5556
15:10:38 INFO ui_util: Initializing UI for MainWindow
15:10:38 INFO thumbnail: Starting thumbnail server listening on port 45013
15:10:38 INFO webengine: WebEngine backend initializing
15:10:39 INFO transition_model: updating transitions model.
15:10:39 INFO effects_model: updating effects model.
15:10:39 INFO emoji_model: updating emoji model.
15:10:39 INFO main_window: InitCacheSettings
15:10:39 INFO main_window: cache-mode: CacheMemory
15:10:39 INFO main_window: cache-limit-mb: 250
15:10:39 INFO main_window: cache-ahead-percent: 0.7
15:10:39 INFO main_window: cache-preroll-min-frames: 24
15:10:39 INFO main_window: cache-preroll-max-frames: 48
15:10:39 INFO main_window: cache-max-frames: 600
15:10:39 INFO main_window: Creating CacheMemory object with 262144000 byte limit
15:10:39 INFO preview_thread: QThread Start Method Invoked
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/thumbnail
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/blender
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/title
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO app: Process command-line arguments: ['/home/user/Video/wana/wana.osp']
15:10:39 INFO settings: reducing file: /home/user/Video/wana/wana.osp to dir: /home/user/Video/wana
15:10:39 INFO preview_thread: Detected default audio device sample rate: 44100
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/thumbnail
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/blender
15:10:39 INFO main_window: Cleared temporary files: /home/user/.openshot_qt/title
15:10:39 INFO project_data: Setting default profile to HD 720p 30 fps
15:10:39 INFO project_data: Apply default audio playback settings: 44100, 2 channels
15:10:39 INFO project_data: Loading project file: /home/user/Video/wana/wana.osp
15:10:39 ERROR json_data: Couldn't load project data file
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/openshot_qt/classes/json_data.py", line 208, in read_from_file
    return json.loads(contents)
  File "/usr/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO project_data: Successfully loaded legacy project file: /home/user/Video/wana/1465690070-aa32bf6348a9388c78905ddefa7019f2.720x368.jpg
15:10:39 INFO project_data: checking project files...
15:10:39 INFO project_data: checking file /home/user/Video/wana/1465690070-aa32bf6348a9388c78905ddefa7019f2.720x368.jpg
15:10:39 INFO project_data: checking file /home/user/Video/wana/We Are Not Alone in the Universe (Rework 2013].mp3
15:10:39 INFO project_data: Project data: openshot 3.0.0, libopenshot 0.3.0
15:10:39 INFO project_data: Apply default audio playback settings: 44100, 2 channels
15:10:39 INFO version: Found current version: {'trans_rate_stable': 0.01, 'error_rate_unstable': 0.05, 'trans_rate_unstable': 0.001, 'openshot_version': '3.0.0', 'error_rate_stable': 0.5}
15:10:39 INFO sentry: Sentry initialized for 'production': 0.5 sample rate, 0.01 transaction rate
15:10:39 INFO timeline: Adjusting max size of preview image: PyQt5.QtCore.QSize(884, 496)
15:10:39 INFO video_widget: Load: Set video widget display aspect ratio to: 1.7777777910232544
15:10:39 INFO video_widget: Set video widget pixel aspect ratio to: 1.0
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO main_window: updateStatusChanged
15:10:39 INFO timeline: Adjusting max size of preview image: PyQt5.QtCore.QSize(1048, 498)
15:10:39 INFO main_window: Loaded project /home/user/Video/wana/wana.osp
15:10:39 INFO webengine: Registering WebChannel connection with WebEngine
15:10:40 INFO webview: Qt Ready
15:10:40 INFO webview: Angular Ready
15:10:40 ERROR webengine: controllers.js@L1442: Uncaught TypeError: previous_object.splice is not a function
15:10:47 INFO main_window: ---------------- Shutting down -----------------
15:10:47 INFO thumbnail: Shutting down thumbnail server: ('127.0.0.1', 45013)
15:10:48 WARNING logger_libopenshot: Socket operation on non-socket
15:10:48 INFO updates: Remove listener from UpdateManager: <windows.preview_thread.PreviewParent object at 0x7f09307e0280>
15:10:48 INFO logger: Screen eDP-1-1
   devicePixelRatio: 1.0
   logicalDotsPerInch: 96.0
   physicalDotsPerInch: 141.95107844318593
   availableSizes: PyQt5.QtCore.QSize(1876, 1048)
15:10:48 INFO app:             OpenShot's session ended            
15:10:48 INFO app:             Sat Jan 28 15:10:48 2023            
15:10:48 INFO app: ================================================
```

</details>

libopenshot.log: File too small or empty (not uploaded)

**Exception / Stacktrace**
```No stacktrace found in log files```

**Screenshots** (Optional)
If applicable, add screenshots to help explain your problem. You can include screenshots by
copy/pasting them on GitHub or dragging-and-dropping into the GitHub page. All images are public,
so please don't post screenshots containing personal information.

**Comment by Colorjet3** [comment]
What version did you create your project in (2.4.x, 2.5.x, 2.6.x)?  Depending on which version you may not be able to open it in v3.0.0 without having to go through intermediate versions (upgrade from 2.4.x to 2.5.x, etc.).

Also, which specific effect stopped working?

**Comment by Colorjet3** [comment]
Hello @Flashwalker I have not received an update on this.  I am closing the ticket for now.  Please feel free to reopen if you still need assistance.

**Comment by Flashwalker** [comment]
I've been able to open and render the project in v.1.44~alpha1 on my old desktop
but not been able to open in openshot-qt version 2.1. 
There was an error while opening:
```
project_data:INFO Setting default profile to HDV 720 24p
   json_data:ERROR Couldn't load project data file: 'ascii' codec can't decode byte 0xd0 in position 2193: ordinal not in range(128)
project_data:ERROR Failed to load project file /home/user/Видео/wana/wana.osp: 'ascii' codec can't decode byte 0xd0 in position 0: ordinal not in range(128)
 main_window:ERROR Couldn't open project /home/user/Видео/wana/wana.osp
```
<details>

```
$ openshot-qt
Loaded modules from installed directory: /usr/lib/python3/dist-packages/openshot_qt
      launch:INFO ------------------------------------------------
      launch:INFO    OpenShot (version 2.1.0)
      launch:INFO ------------------------------------------------
         app:INFO openshot-qt version: 2.1.0
         app:INFO libopenshot version: 0.1.2
         app:INFO platform: Linux-3.16.0-30-generic-i686-with-LinuxMint-17-qiana
         app:INFO processor: i686
         app:INFO machine: i686
         app:INFO python version: 3.4.3
         app:INFO qt5 version: 5.2.1
         app:INFO pyqt5 version: 5.2.1
    language:INFO Qt Detected Languages: ['ru-RU']
    language:INFO LANG Environment Variable: ru_RU.UTF-8
    language:INFO LOCALE Environment Variable: ru_RU
    language:INFO Attempting to load qt_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Successfully loaded qt_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Attempting to load qtbase_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Successfully loaded qtbase_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Attempting to load qt_ru_RU.UTF-8 in '/usr/lib/python3/dist-packages/openshot_qt/locale/QT'
    language:INFO Attempting to load qt_ru in '/usr/lib/python3/dist-packages/openshot_qt/locale/QT'
    language:INFO Attempting to load qtbase_ru_RU.UTF-8 in '/usr/lib/python3/dist-packages/openshot_qt/locale/QT'
    language:INFO Attempting to load qtbase_ru in '/usr/lib/python3/dist-packages/openshot_qt/locale/QT'
    language:INFO Attempting to load ru_RU.UTF-8/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Attempting to load ru/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Successfully loaded ru/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Exiting translation system (since we successfully loaded: ru_RU.UTF-8)
project_data:INFO Setting default profile to HDV 720 24p
         app:INFO Setting font to /usr/lib/python3/dist-packages/openshot_qt/images/fonts/Ubuntu-R.ttf
logger_libopenshot:INFO Connecting to libopenshot with debug port: 5556
    language:INFO Attempting to load qt_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Successfully loaded qt_ru_RU.UTF-8 in '/opt/qt55/translations'
    language:INFO Attempting to load ru_RU.UTF-8/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Attempting to load ru/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
    language:INFO Successfully loaded ru/LC_MESSAGES/OpenShot in '/usr/lib/python3/dist-packages/openshot_qt/locale'
link SVGID_18_-5 hasn't been detected!
link SVGID_18_-5 hasn't been detected!
link SVGID_22_-5 hasn't been detected!
link SVGID_24_-2 hasn't been detected!
link SVGID_24_-2 hasn't been detected!
     ui_util:INFO Initializing UI for MainWindow
     ui_util:INFO Binding event MainWindow:actionNew_trigger
     ui_util:INFO Binding event MainWindow:actionOpen_trigger
     ui_util:INFO Binding event MainWindow:actionSave_trigger
     ui_util:INFO Binding event MainWindow:actionUndo_trigger
     ui_util:INFO Binding event MainWindow:actionSaveAs_trigger
     ui_util:INFO Binding event MainWindow:actionImportFiles_trigger
     ui_util:INFO Binding event MainWindow:actionImportImageSequence_trigger
     ui_util:INFO Binding event MainWindow:actionRedo_trigger
     ui_util:INFO Binding event MainWindow:actionRemoveClip_trigger
     ui_util:INFO Binding event MainWindow:actionRemoveTransition_trigger
     ui_util:INFO Binding event MainWindow:actionExportVideo_trigger
     ui_util:INFO Binding event MainWindow:actionUploadVideo_trigger
     ui_util:INFO Binding event MainWindow:actionAddTrack_trigger
     ui_util:INFO Binding event MainWindow:actionPreferences_trigger
     ui_util:INFO Binding event MainWindow:actionPlay_trigger
     ui_util:INFO Binding event MainWindow:actionJumpStart_trigger
     ui_util:INFO Binding event MainWindow:actionRewind_trigger
     ui_util:INFO Binding event MainWindow:actionFastForward_trigger
     ui_util:INFO Binding event MainWindow:actionJumpEnd_trigger
     ui_util:INFO Binding event MainWindow:actionArrowTool_trigger
     ui_util:INFO Binding event MainWindow:actionSnappingTool_trigger
     ui_util:INFO Binding event MainWindow:actionAddMarker_trigger
     ui_util:INFO Binding event MainWindow:actionPreviousMarker_trigger
     ui_util:INFO Binding event MainWindow:actionNextMarker_trigger
     ui_util:INFO Binding event MainWindow:actionFilesShowAll_trigger
     ui_util:INFO Binding event MainWindow:actionFilesShowVideo_trigger
     ui_util:INFO Binding event MainWindow:actionFilesShowAudio_trigger
     ui_util:INFO Binding event MainWindow:actionFilesShowImage_trigger
     ui_util:INFO Binding event MainWindow:actionTransitionsShowAll_trigger
     ui_util:INFO Binding event MainWindow:actionTransitionsShowCommon_trigger
     ui_util:INFO Binding event MainWindow:actionEffectsShowAll_trigger
     ui_util:INFO Binding event MainWindow:actionEffectsShowVideo_trigger
     ui_util:INFO Binding event MainWindow:actionEffectsShowAudio_trigger
     ui_util:INFO Binding event MainWindow:actionTimelineZoomIn_trigger
     ui_util:INFO Binding event MainWindow:actionTimelineZoomOut_trigger
     ui_util:INFO Binding event MainWindow:actionTitle_trigger
     ui_util:INFO Binding event MainWindow:actionAnimatedTitle_trigger
     ui_util:INFO Binding event MainWindow:actionFullscreen_trigger
     ui_util:INFO Binding event MainWindow:actionAbout_trigger
     ui_util:INFO Binding event MainWindow:actionThumbnailView_trigger
     ui_util:INFO Binding event MainWindow:actionDetailsView_trigger
     ui_util:INFO Binding event MainWindow:actionReportBug_trigger
     ui_util:INFO Binding event MainWindow:actionAskQuestion_trigger
     ui_util:WARNING Icon theme im-message-new not found. Will use backup icon.
     ui_util:INFO Binding event MainWindow:actionTranslate_trigger
     ui_util:INFO Binding event MainWindow:actionDonate_trigger
     ui_util:INFO Binding event MainWindow:actionHelpContents_trigger
     ui_util:INFO Binding event MainWindow:actionSimple_View_trigger
     ui_util:INFO Binding event MainWindow:actionAdvanced_View_trigger
     ui_util:INFO Binding event MainWindow:actionFreeze_View_trigger
     ui_util:WARNING Icon theme locked not found. Will use backup icon.
     ui_util:INFO Binding event MainWindow:actionUn_Freeze_View_trigger
     ui_util:WARNING Icon theme locked not found. Will use backup icon.
     ui_util:INFO Binding event MainWindow:actionShow_All_trigger
     ui_util:INFO Binding event MainWindow:actionProfile_trigger
     ui_util:INFO Binding event MainWindow:actionAdd_to_Timeline_trigger
     ui_util:INFO Binding event MainWindow:actionPreview_File_trigger
     ui_util:INFO Binding event MainWindow:actionRemove_from_Project_trigger
     ui_util:INFO Binding event MainWindow:actionFile_Properties_trigger
     ui_util:INFO Binding event MainWindow:actionRemoveTrack_trigger
     ui_util:INFO Binding event MainWindow:actionRemoveMarker_trigger
     ui_util:INFO Binding event MainWindow:actionAddTrackAbove_trigger
     ui_util:INFO Binding event MainWindow:actionAddTrackBelow_trigger
     ui_util:INFO Binding event MainWindow:actionRemoveEffect_trigger
     ui_util:INFO Binding event MainWindow:actionSplitClip_trigger
     ui_util:INFO Binding event MainWindow:actionProperties_trigger
     ui_util:INFO Binding event MainWindow:actionRenameTrack_trigger
     ui_util:INFO Binding event MainWindow:actionUpdate_trigger
     ui_util:INFO Binding event MainWindow:actionTutorial_trigger
     ui_util:WARNING Icon theme im-message-new not found. Will use backup icon.
     ui_util:INFO Binding event MainWindow:actionAnimation_trigger
     ui_util:INFO Binding event MainWindow:actionLockTrack_trigger
     ui_util:INFO Binding event MainWindow:actionUnlockTrack_trigger
     version:ERROR Failed to get version from: http://www.openshot.org/version/json/
 files_model:INFO updating files model.
transition_model:INFO updating transitions model.
timeline_webview:INFO Qt Found!
timeline_webview:INFO $scope.Qt = true;
timeline_webview:INFO UpdateLayerIndex
effects_model:INFO updating effects model.
effects_model:INFO category: Video
effects_model:INFO category: Video
effects_model:INFO category: Video
effects_model:INFO category: Video
effects_model:INFO category: Video
effects_model:INFO category: Video
effects_model:INFO category: Video
properties_model:INFO updating clip properties model.
preview_thread:INFO QThread Start Method Invoked
 main_window:INFO Clear all thumbnails: /home/user/.openshot_qt/thumbnail
preview_thread:INFO initPlayer
 main_window:INFO Clear all animations: /home/user/.openshot_qt/blender
 main_window:INFO Clear all backups: /home/user/.openshot_qt/backup
project_data:INFO Setting default profile to HDV 720 24p
preview_thread:INFO refreshFrame
preview_thread:INFO self.player.Position(): 1
timeline_webview:INFO UpdateLayerIndex
 main_window:INFO updateStatusChanged
    tutorial:INFO process tutorial dialogs
         app:INFO Process command-line arguments: ['/usr/bin/openshot-qt']
 main_window:INFO updateStatusChanged
preview_thread:INFO onModeChanged
properties_model:INFO Update item: 
properties_model:INFO updating clip properties model.
timeline_webview:INFO redraw_audio_onTimeout
     ui_util:INFO Initializing UI for dlgPreferences
 main_window:INFO Preferences add cancelled
 main_window:INFO Clear all thumbnails: /home/user/.openshot_qt/thumbnail
 main_window:INFO Clear all animations: /home/user/.openshot_qt/blender
 main_window:INFO Clear all backups: /home/user/.openshot_qt/backup
project_data:INFO Setting default profile to HDV 720 24p
   json_data:ERROR Couldn't load project data file: 'ascii' codec can't decode byte 0xd0 in position 2193: ordinal not in range(128)
project_data:ERROR Failed to load project file /home/user/Видео/wana/wana.osp: 'ascii' codec can't decode byte 0xd0 in position 0: ordinal not in range(128)
 main_window:ERROR Couldn't open project /home/user/Видео/wana/wana.osp
link SVGID_18_-5 hasn't been detected!
link SVGID_18_-5 hasn't been detected!
link SVGID_22_-5 hasn't been detected!
link SVGID_24_-2 hasn't been detected!
link SVGID_24_-2 hasn't been detected!
     ui_util:INFO Initializing UI for Dialog
 main_window:INFO About Openshot add cancelled
preview_thread:INFO exiting thread
preview_thread:INFO onModeChanged

```
</details>

**Comment by Flashwalker** [comment]
> Hello @Flashwalker I have not received an update on this. I am closing the ticket for now. Please feel free to reopen if you still need assistance.

can't reopen

**Comment by Flashwalker** [comment]
> Also, which specific effect stopped working?

There was these effects:
Old Film
White Balance
Saturation
<details>
<summary>Screenshot</summary>

![2023-02-16_07-59-58_1920x1080_scrot](https://user-images.githubusercontent.com/2188215/219318787-bf785ecb-e8db-4b19-90d5-04604b828c8b.jpg)

</details>

**Comment by Colorjet3** [comment]
Hello @Flashwalker.  Those effects don't even exist or they have changed enough that you are going to have these challenges jumping from an project that was created 5-6 version before and then expect it to open.  It will be best if you recreate your project in v3.0.0.

**Comment by Flashwalker** [comment]
Should i recreate it after a few releases again?

**Comment by Colorjet3** [comment]
No, you should not have to recreate your project every time you upgrade.  As long as you keep up with the Openshot upgrades I recommend that you open your existing projects and ensure that they open and preview properly.  You can even go as far performing a "Save Project As" to be sure.

At this time, Openshot is receiving many fixes and enhancements within the v3.0.0 release of Daily Builds.  You can find these here:  openshot.org/download/#daily.

I don't expect issues with newer projects that were created with v2.6.1+ and working with them in v3.0.x series at this time.
